src/seer/automation/codegen/models.py

```diff
@@ -194,7 +194,16 @@ class Comment(BaseModel):
         start_line: int
         suggestion: str | None = None
 
+    class PrDescription(BaseModel):
+        purpose: str
+        key_technical_changes: str
+        architecture_decisions: str
+        dependencies_and_interactions: str
+        risk_considerations: str
```

Comment:
Consider adding field validators to ensure non-empty strings and reasonable length limits for the PR description fields. This will help maintain quality of the generated descriptions and prevent potential issues with very long content.
```suggestion
class PrDescription(BaseModel):
    purpose: str = Field(..., min_length=1, max_length=2000)
    key_technical_changes: str = Field(..., min_length=1, max_length=2000)
    architecture_decisions: str = Field(..., min_length=1, max_length=2000)
    dependencies_and_interactions: str = Field(..., min_length=1, max_length=2000)
    risk_considerations: str = Field(..., min_length=1, max_length=2000)
    notable_implementation_details: str = Field(..., min_length=1, max_length=2000)
```